スーパーで買い物プログラム

Kumagai6824 opened the pull request:
プログラム内容はREADMEに記載しましたのでご確認お願いいたします！
前回からのアドバイス内容で今いちわかってないところ、気になる点は以下です。

①第四回課題にて、小山さんより.gitignore内で「改行をいれましょう」とのコメントをいただきましたが、どこを改行するのか分かっておらず、今回は取り合えず、一番最後の行を改行してみました。

②Mainの`Arrays.stream(Category.values()).toList().forEach(Category -> {`　以降に処理を詰め込みすぎてしまった気がしていますが、どうでしょうか。本当は可読性をもっと追及した方がいいのかもしれないですが、現段階で頭がいっぱいいっぱいだったので、もっと整理できる方法があれば、ヒントをいただきたいです。

アドバイスよろしくお願いします！

Kumagai6824 committed:
[Modified]README.md

Kumagai6824 committed:
[ADD]Main/Category/NoSuchKeyException/Products

TomooAkamatsu commented on src/Products.java:
```diff
@@ -0,0 +1,25 @@
+public class Products {
```
このクラスのインスタンスは一つの商品を表すと思うので、複数形のsは要らないかなと思いました

TomooAkamatsu commented on src/Products.java:
```diff
@@ -0,0 +1,25 @@
+
+    private final String productName;
+    private final Category productCategory;
+    private final int productCost;
```
productクラスのフィールドなので変数名の頭にproductはつけなくて良いかなと思います
getterも合わせてお願いします！

TomooAkamatsu commented on src/Main.java:
```diff
@@ -0,0 +1,92 @@
+                        pickedProducts.add(chosenProduct);
+                        flag = false;
+                    } else {
+                        throw new NoSuchKeyException("You can only use <" + rangeMin + "> to <" + rangeMax + ">");
```
ちゃんと制御できていて良いですね！

TomooAkamatsu commented on src/Main.java:
```diff
@@ -0,0 +1,92 @@
+    public static void main(String[] args) {
+
+        //スーパーの商品リストを作る
+        Map<Integer, Products> productsList = new HashMap<>();
```
商品リストなのでListとつけたいところですが、中身はMap型なのでListという命名は避けた方がいいですね。

Kumagai6824 replied:
マップのキーと商品情報ということで、keysToProductsとしてみました。

TomooAkamatsu commented on src/Main.java:
```diff
@@ -0,0 +1,92 @@
+                    rangeMax = 9;
+            }
+
+            boolean flag;
```
細かいですが、ここでtrueと入れておいて、catchの部分のtrueはなくても良いかもです。
好みの問題な気がするので、一応コメントはしますが気にしないでください笑

Kumagai6824 replied:
ご指摘ありがとうございます！これで1文減らせるので、対応してみました。

TomooAkamatsu commented:
> 今回は取り合えず、一番最後の行を改行してみました。

各ファイルの最終行という認識でOKです！
セーブ時に自動で改行を入れてくれる設定があるので調べてみて下さい。

>以降に処理を詰め込みすぎてしまった気がしていますが、どうでしょうか。

個人的には特に読みづらいとは思いませんでした！
が、私よりもっと経験のある方の意見も聞きたいですね笑
機能毎にメソッドに切り出せる部分があれば、そのほうが良いかなとは思います。

Kumagai6824 commented:
> 各ファイルの最終行という認識でOKです！
 セーブ時に自動で改行を入れてくれる設定があるので調べてみて下さい。

設定確認できました！

> 機能毎にメソッドに切り出せる部分があれば、そのほうが良いかなとは思います。

今後のプログラムを書く時に意識したいと思います！

Kumagai6824 committed:
[Modified]
Main:Changed boolean flag part.
Product:Changed class name from Products. Changed name of variables.
